src/components/transactions/SignTxButton/index.tsx

```diff
@@ -11,6 +11,31 @@ import IconButton from '@mui/material/IconButton'
 import CheckIcon from '@mui/icons-material/Check'
 import Track from '@/components/common/Track'
 import { TX_LIST_EVENTS } from '@/services/analytics/events/txList'
+import { txSubscribe, TxEvent } from '@/services/tx/txEvents'
+
+const useIsSignatureProposalPending = (txSummary: TransactionSummary) => {
+  const wallet = useWallet()
+  const isSignable = isSignableBy(txSummary, wallet?.address || '')
+  const [isSignatureProposalPending, setIsSignatureProposalPending] = useState<boolean>(false)
+
+  useEffect(() => {
+    if (isSignable) {
```

Comment:
Maybe we should listen to propose failed instead?

Reply:
We set the signature proposal as pending [after a successful proposal has happened](https://github.com/safe-global/web-core/blob/main/src/services/tx/txSender.ts#L147-L157).

Reply:
I've now optimised the hook and added a comment.